Parameters

jburel opened the pull request:
Add `JAVAVER` parameters. Default is OpenJDK 1.8 across the various installations. This was not the case.
The first 6 commits were tested by travis in PR https://github.com/ome/omero-install/pull/83
Re-organise step installation dependencies.
Update the autogenerate script.

Also Add `PGVER` (for postgres) and `ICEVER` (for ice)
cc @hflynn

jburel committed:
Add support for new JAVAVER argument

jburel committed:
Add support for oracle java 1.7/1.8 openjdk 1.7/1.8 on centOS

jburel committed:
Add support for oracle Java and openjdk 1.7/18 ubuntu

jburel committed:
Add support for oracle Java 1.7/1.8

jburel committed:
Install openjdk 1.8 on debian/jessie

jburel committed:
Make Java openjdk 1.8 the default.

jburel committed:
Re-organise steps to split dependencies intallation.

jburel committed:
Add marker for Java version for generated doc.

jburel committed:
Re-organise steps for ubuntu and debian

jburel committed:
Review doc of dependencies installation

joshmoore commented:
Two quick points:
- I was slightly surprised by the mention of SCL? Is this a leftover from our pre-IUS work?
- As I'm starting to base other repos on this repo, renamings etc. will become breaking changes. Probably manageable for the moment, just something to be aware of.

jburel commented:
@joshmoore:
- At the time of ius work, we decided to keep the scl script but we were not going to expose it in the doc. Happy to remove it we think it is not useful.
- We might have one more "ice" breaking split, but certainly need to be synchronized before going ahead.

jburel committed:
Fix name.

jburel committed:
Rename java version arg: open17/18 to openjdk17/18

jburel committed:
Set the default to openjdk 1.8

jburel commented:
@aleksandra-tarkowska: added option for openjdk17/18-devel as suggested

jburel committed:
Add openjdk17/18-devel option.

jburel commented:
@pwalczysko: I have tested all the java options using travis (list below)
- https://travis-ci.org/ome/omero-install/builds/116131000 openjdk18 (default)
- https://travis-ci.org/ome/omero-install/builds/116145400 openjdk17
- https://travis-ci.org/ome/omero-install/builds/116151515 oracle17
- https://travis-ci.org/ome/omero-install/builds/116206869 oracle18
- https://travis-ci.org/ome/omero-install/builds/116227992 openjdk18-devel
- https://travis-ci.org/ome/omero-install/builds/116239304 openjdk17-devel

jburel committed:
Add icever parameter

jburel committed:
Add ice35-devel option

jburel committed:
Add ice-devel parameter

jburel committed:
Add Ice install to the walkthrough

jburel committed:
Add Postgres parameter

jburel committed:
Install the recommended postgres version.

The default version installed using:
apt-get -y install postgresql is 9.3.
Version marked as deprecated in our requirements page.

jburel commented:
@joshmoore: I have now finished the split. I need to review the content of some files but it should not "break" anymore

jburel committed:
Install postgres

jburel committed:
Add postgres installation to doc.

jburel committed:
Remove leading whitespace

jburel committed:
Add support for postgres 9.5

jburel committed:
Bump pg94 version

jburel committed:
Add sections for new parameters.

jburel committed:
Add support for postgres 9.5 on centos7

jburel committed:
Add support for pg 9.5 ubuntu

atarkowska commented:
@jburel if we already install python-virtualenv why just not create a venv step and

```
virtualenv /home/omero/omero-virtualenv --system-site-packages
pip install --upgrade pip
pip install Pillow
pip install ...
```

instead of line 16 and 19 https://github.com/jburel/omero-install/blob/java-parameters/linux/step01_centos7_deps.sh#L15 ?

joshmoore commented:
@aleksandra-tarkowska : you mean in terms of what we should be suggesting to others to do?

atarkowska commented:
yes, this is healthier way, no?
I thought you suggested requirement files for omero-install, right?

```
pip install -r requirement.txt
```

aleksandra-tarkowska commented:
test

atarkowska committed:
fix path

atarkowska commented:
@jburel I started playing what that, seems that I can reduce dockerfile to

```
RUN JAVAVER=nojava PGVER=nopg sh install_centos7_nginx.sh 
```

I found one issue with settings.env see https://github.com/aleksandra-tarkowska/omero-install/commit/3934c82d6a7dcd5a0ed92a0c154ffb15fef0dac7

otherwise I am getting:

```
+ source settings.env
/tmp/omero-install/linux/install_centos7_nginx.sh: line 8: settings.env: No such file or directory
```

so far so good :+1: 
I will open a PR against your branch once I got that fully working

jburel commented:
@aleksandra-tarkowska: I have not added option like `nojava`, `nopg`.but it will work 
any string not matching the support values will not install anything.
I will probably add that to the readme.

jburel commented:
@aleksandra-tarkowska: requirement file is listed there https://trello.com/c/c7cIQbn2/45-omero-install-new-steps already.
I was focusing on the "breaking" changes first so @joshmoore does not work constantly against a moving target.

jburel committed:
Indicate how not to install Java or Postgres

atarkowska commented:
@jburel random value to skip step works well, I didn't know that wasn't expected :+1:

jburel committed:
Add support pg 9.5

jburel commented:
@aleksandra-tarkowska: for `source.env` comment, there is a step in the install that copy the file.
That's maybe why it was not working for you

jburel commented:
50d892e: I had to rename a file to match the rest. It should be the last one
remaining tasks:
- Introduce requirements files
- fix pg 9.5 installation

jburel committed:
Rename file to match the rest of the dependencies split.

jburel committed:
Synchronize with settings in devslave-c7-docker repo

jburel committed:
Add requirements files.

jburel committed:
Change order

jburel committed:
Use requirements file from OMERO.server to install web dependencies

jburel committed:
Use requirements to install web dependencies (apache)

jburel committed:
Remove web config.

jburel committed:
Enable python27

jburel committed:
Install in virtualenv and do not user requirements files

jburel committed:
Fix apache24 install on centos6 py27 ius.

jburel committed:
Modify reading of requirements. Last line was not read.

jburel committed:
Remove Django from the requirements

This is now installed using the web requirement files.

atarkowska committed:
check if omero user exist

jburel committed:
Do no upgrade to the latest. This creates a locale error

To be investigated.

jburel committed:
Add pg check.

jburel committed:
Install dependencies

atarkowska commented on linux/step01_centos7_ice_deps.sh:
```diff
@@ -0,0 +1,18 @@
+	curl -o /etc/yum.repos.d/zeroc-ice-el7.repo \
+	http://download.zeroc.com/Ice/3.5/el7/zeroc-ice-el7.repo
+
+	yum -y install ice ice-python ice-java-devel ice-servers
```
could you change to `ice-python-devel` please?

jburel replied:
done

jburel committed:
use ice-python-devel if ice35-devel option is on.

jburel committed:
Review docker run to start postgres 9.5

jburel committed:
Clarify how to use postgres option.

jburel committed:
Replace CMD by ENTRYPOINT to pass the pg parameter.

jburel commented:
pg95 option tested using travis: Build https://travis-ci.org/ome/omero-install/builds/117091998
pg94 (default) https://travis-ci.org/ome/omero-install/builds/117113482

jburel committed:
Upgrade pip.

jburel committed:
Do not upgrade pip if docker.

jburel committed:
Match change done for nginx version.

jburel committed:
Fix indent.

jburel committed:
Align with code changes

jburel committed:
Combine files.

jburel committed:
Update the install scripts.

jburel committed:
Merge file.

jburel committed:
Fix doc generation for omero install.

jburel committed:
Fix section to install pg

jburel committed:
Fix omero installation

jburel committed:
Fix user creation display in doc.

jburel committed:
Use same file to install omero.

jburel committed:
Add omego to the list requirements.

jburel committed:
No longer specify parameter.

jburel committed:
Split step04.

jburel committed:
Match file split in doc.

atarkowska commented on linux/step04_all_omero.sh:
```diff
@@ -2,24 +2,43 @@
-unzip -q server-ice35.zip
+#start-install
+if [ $OMEROVER = omerodev ]; then
+	/home/omero/omeroenv/bin/omego download --branch OMERO-DEV-latest server
```
@jburel is there any reason why we just cannot have `/home/omero/omeroenv/bin/omego download --branch $OMEROVER server`. that will simplify a lot. This file is not usable for me at the moment

manics replied:
:+1: If you set a default for OMEROVER (`OMEROVER=${OMEROVER:-latest}`) you can get rid of the conditional, since `--release` is a synonym for `--branch`

joshmoore replied:
^ /me didn't know this. I'll guess this falls under https://github.com/ome/omego/issues/27

manics replied:
I think this is one of those arguments that can't be simplified, since I made them synonyms for convenience (such as the case above), even though `branch` and `release` have different meanings.

jburel committed:
Cap omego version to 0.3.0

jburel committed:
Change OMEROVER parameter.

jburel committed:
Remove testing code

jburel committed:
Update readme

jburel commented:
Probably time to draw the line, further improvements can be done in follow-up PRs

jburel committed:
Match scripts change.

atarkowska commented on linux/step04_all_omero.sh:
```diff
@@ -23,16 +23,9 @@ if [[ ! $PY_ENV = "py27_ius" ]]; then
+#start-release
+/home/omero/omeroenv/bin/omego download --branch $OMEROVER server
+#end-release
 
```
This is great, ideally would be to deactivate virtualenv after that step

pwalczysko commented:
Sequence of commands 
- `PGVER=pg95 ICEVER=ice35-devel JAVAVER=oracle18 ./docker-build.sh centos6_apache22`
- executed successfully but then

```
[pwalczysko@ls31619 ~/Work/omero-install/linux/test]$ docker run --rm -it -p 8080:80 -p 4063:4063 -p 4064:4064 omero_install_test_centos6_apache22
postgresql-9.4: unrecognized service
```

And no further output came. Is this expected ?

jburel commented:
if you use postgres 9.5
the command to run  is:
`docker run --rm -it -p 8080:80 -p 4063:4063 -p 4064:4064 omero_install_test_centos6_apache22 pg95`
cf. readme

pwalczysko commented:
This is clear, and my fault. But reading the README again, is there any indication that this is a compulsory parameter specific for centos6 ? See below, for ubuntu there is no suffix pg95 mentioned...

```
For example:

    PGVER=pg95 ./docker-build.sh centos6_py27_ius_nginx
    docker run --rm -it -p 8080:80 -p 4063:4063 -p 4064:4064 omero_install_test_centos6_py27_ius_nginx pg95

    PGVER=pg95 ./docker-build.sh ubuntu1404_nginx
    docker run --rm -it -p 8080:80 -p 4063:4063 -p 4064:4064 omero_install_test_ubuntu1404_nginx
```

jburel commented:
It is only required for centos
i.e. `service postgresql-9.4 start`
and on ubuntu `service postgresql start`

pwalczysko commented:
I would suggest then to change the `For example` in the README to say `Do this and that for all centos..., do this and that for ubuntu`

pwalczysko commented:
So the `PGVER=pg95 ICEVER=ice35-devel` option worked for centos6_apache22. Should I go through the whole list of the supported OS-es with those two versions of PG and ice ? Is this a good plan ?

jburel commented:
I have already tested the pg version with travis cf. https://github.com/ome/omero-install/pull/94#issuecomment-198709823 so probably not required.
For the ice version, I have put the structure in place and @aleksandra-tarkowska has tested the devel option
so probably not required to go over the manual testing.
I will update the readme

jburel committed:
Add postgres check.

jburel committed:
Clarify how to run image if pg version not the default one.

pwalczysko commented:
Tested:
`PGVER=pg94 ./docker-build.sh centos6_py27_nginx` followed by
`docker run --rm -it -p 8080:80 -p 4063:4063 -p 4064:4064 omero_install_test_centos6_py27_nginx pg94`
and 
`PGVER=pg95 ./docker-build.sh centos6_py27_nginx` followed by
`docker run --rm -it -p 8080:80 -p 4063:4063 -p 4064:4064 omero_install_test_centos6_py27_nginx pg95` 
both went fine.
I also used the README to construct the above commands and it led me well, so this is fine as well.
All good here.

atarkowska commented:
@pwalczysko what about centos7?

pwalczysko commented:
@aleksandra-tarkowska : Will have a look.

pwalczysko commented:
Could we replace in the README the lines

```
./docker-build.sh centos7
        CID=$(docker run -d ... -v /sys/fs/cgroup:/sys/fs/cgroup:ro omero_install_test_centos7)
        #CID=$(docker run -d ... --privileged omero_install_test_centos7)
```

by

```
./docker-build.sh centos7
        #CID=$(docker run -d -p 2222:22 -p 8080:80 -p 4063:4063 -p 4064:4064  --v /sys/fs/cgroup:/sys/fs/cgroup:ro omero_install_test_centos7)
        CID=$(docker run -d -p 2222:22 -p 8080:80 -p 4063:4063 -p 4064:4064 --privileged omero_install_test_centos7)
```

simply spelling out the ... ? This costs me a bit of effort every time to figure out what should be actually in.

EDIT: Also, the next line does not work for me, the 
`ssh -p 2222 omero@<address of container>`  does.

EDIT 2: Mention that you have to manually start the server and the webserver in this centos7 workflow.

manics commented:
To reduce duplication in the readme you could do what @pwalczysko suggests but store the common args in a variable instead, e.g.

```
COMMON_ARGS="-p 2222:22 -p 8080:80 -p 4063:4063 -p 4064:4064"

CID=$(docker run -d $COMMON_ARGS -v /sys/fs/cgroup:/sys/fs/cgroup:ro omero_install_test_centos7)
```

pwalczysko commented:
Running:
`./docker-build.sh centos7`
`CID=$(docker run -d -p 2222:22 -p 8080:80 -p 4063:4063 -p 4064:4064 --privileged omero_install_test_centos7)`
`ssh -p 2222 root@<address of container>`
`cd /omero-install-test`
`PGVER=pg95 ICEVER=ice35-devel JAVAVER=oracle18 bash install_centos7_nginx.sh`
ended up with

```
...
+ systemctl daemon-reload
+ systemctl start postgresql-9.5.service
Job for postgresql-9.5.service failed because the control process exited with error code. See "systemctl status postgresql-9.5.service" and "journalctl -xe" for details.
[root@3d27f3426fdd omero-install-test]# 
```

EDIT: Did not repeat, seems fine.

pwalczysko commented:
Rerun the commands from the previous comment https://github.com/ome/omero-install/pull/94#issuecomment-200928114 once more and all works fine. I think it is safe to assume that this settings work, as the first time in https://github.com/ome/omero-install/pull/94#issuecomment-200928114 I was running some other build in the same container prior to the command itself.

pwalczysko commented:
Also on Centos7, `PGVER=pg94 ICEVER=ice35 JAVAVER=openjdk17 bash install_centos7_nginx.sh` worked as expected. 
This means on Centos7 we have tested
- both possible values of PGVER
- both possible values of ICEVER
- two values of JAVAVER 
  - openjdk17 and 
  - oracle18

Should I go through all the other 4 JAVAVER options as well @jburel ?

jburel commented:
@pwalczysko: no need. see https://github.com/ome/omero-install/pull/94#issuecomment-197047558

atarkowska commented:
@jburel all the improvement is great, definitely more work needed, ideally if in the next round we will get virtualenv for centos7
I made a progress on devspace. see what kind of adjustments we need https://github.com/joshmoore/devspace/pull/1/files#diff-142bdc956123e5b32d1d777e99af6905R43

jburel commented:
@aleksandra-tarkowska: I have started work on virtual env, I see that more as "breaking chance" in the doc so probably better to have it for 5.3.

jburel committed:
Clarify centos7 section

jburel committed:
Swap commented out line

atarkowska commented on linux/step01_centos7_deps.sh:
```diff
@@ -1,49 +1,18 @@
 
-systemctl start postgresql-9.4.service
-systemctl enable postgresql-9.4.service
+pip install -r requirements.txt
```
would it be possible to get absolute patch here? I am guessing ``dirname $0`/`

jburel replied:
done

jburel committed:
Add path.

pwalczysko commented:
Following the README, I ran on Centos7  `JAVAVER=oracle17 ICEVER=ice35-devel bash install_centos7_apache24.sh`. The README led me through as expected and the installation worked as well.
All good here.

atarkowska commented on linux/setup_omero_db.sh:
```diff
@@ -0,0 +1,8 @@
+
+set -e -u -x
+
+source settings.env
```
path please

atarkowska commented on linux/setup_omero_nginx.sh:
```diff
@@ -0,0 +1,9 @@
+
+set -e -u -x
+
+source settings.env
```
the same path

jburel replied:
done

jburel committed:
Swap order (prep for ice 36 build)

jburel committed:
Add path.

jburel commented:
All:
I have started in another branch (on top of that one) to work on the ice3.6 support.If we do not need further changes it might be good to merge.

jburel committed:
Remove search ice.

atarkowska commented:
@pwalczysko are you happy with all the changes? Could this PR being merged as is blocking me

pwalczysko commented:
I was happy before https://github.com/ome/omero-install/pull/94#issuecomment-201232444. Are the 3 new commits tested by Travis ? If yes, happy to merge.

jburel commented:
@pwalczysko: 2/3 are tested by travis and `149c8a4` tested by @aleksandra-tarkowska

pwalczysko commented:
Happy then, thank you.

atarkowska commented:
This was successfully used in https://github.com/openmicroscopy/devspace/pull/20.

atarkowska commented on linux/step05_centos7_nginx.sh:
```diff
@@ -1,15 +1,19 @@
+#end-copy
+
 #start-install
 yum -y --enablerepo=cr install nginx
```
why do we need CR? EPEL is installed in https://github.com/ome/omero-install/pull/94/files#diff-2722b62c943402fa409b2ef40375cf5dR3

from EPEL you also install nginx/1.6.3

aleksandra-tarkowska replied:
also this is not consistent with C6 where we manually install from nginx website. Is that expected?

jburel replied:
I will have to look into it, since we have been splitting several files

jburel replied:
cr is not required. so it should be removed. The version installed will be 1.6.3. 
To match the recommended version cf. https://github.com/openmicroscopy/ome-documentation/pull/1439, we will have to follow the same approach that the one used for centos 6. 
For the rest of the dependencies, we have installed the recommended version
I will update the code according

atarkowska replied:
@jburel I am not sure if I follow but for C6, changing to EPEL will install nginx  1.0.15. I am guessing is better to leave as it is (installing from Nginx repository)

jburel replied:
for C6, yes the version is old.

We recommend version in our requirements page for java, ice, pg etc. in the doc.
The version installed in the installation script should match what is recommended, for nginx we recommend 1.8/1.9 so that what we should/install test.
That's another reason why we are also thinking of moving the sysadmin doc under omero-install so we will hopefully keep things in synch (it has not been the case e.g pg on ubuntu!)

note that I am currently fixing the ubuntu install script since I just noticed that 1.4.6 is currently installed.

jburel committed:
Install the recommended version of nginx.

jburel commented:
@aleksandra-tarkowska: nginx version installed should now be 1.8+ for centos6/7 and ubuntu1404/debian8 cf. https://github.com/openmicroscopy/ome-documentation/pull/1439

jburel committed:
Install more recent version of nginx

previously 1.4.6 now 1.8.1

jburel committed:
Add documentation.

jburel committed:
Add debian8 specific file to install nginx.

jburel committed:
Reflect changes in script.

jburel commented:
nginx changes have been tested via travis.
Bumping debian to 1.8 will happen in another PR since it requires some web adjustments (some folders no longer exists e.g. site-available in version 1.8)

atarkowska commented:
@jburel there is one more file that need to be fixed https://github.com/ome/omero-install/blob/develop/linux/omero-systemd.service

> This means on Centos7 we have tested
> - both possible values of PGVER

@pwalczysko how did you manage to start server with PostgreSQL 9.5 if pg version is hardcoded in daemon script ?

jburel commented:
@aleksandra-tarkowska: Thanks I thought I fixed it. I will try to do that today

pwalczysko commented:
> @pwalczysko how did you manage to start server with PostgreSQL 9.5 if pg version is hardcoded in daemon script ?

The comment of mine just claims I have tried both PGVER values. Did not check the Postgres version truly installed.

pwalczysko commented:
> @pwalczysko how did you manage to start server with PostgreSQL 9.5 if pg version is hardcoded in daemon script ?

The comment of mine just claims I have tried both PGVER values. Did not check the Postgres version truly installed.

joshmoore commented:
> I will try to do that today

@jburel: Thanks. Let us know though, that could also be a follow on PR.

sbesson commented:
Since this is also currently consumed (and someway tested) via https://github.com/openmicroscopy/devspace/pull/20/files#diff-91f78839f1da3c73c6a67587b59bc8a5R18, I would be in favor of getting this merged and point to the organization repository for the container installation.

jburel commented:
I did not have time yesterday probably better to merge this PR and fix  it in a follow-up PR so we can move forward on other fronts

sbesson commented:
Merging to unlock devspace.
